tools/accuracy_checker/accuracy_checker/adapters/retinaface.py

```diff
@@ -43,50 +50,78 @@ def process(self, raw, identifiers=None, frame_meta=None):
             x_mins, y_mins, x_maxs, y_maxs = np.array(proposals_list).T # pylint: disable=E0633
             detection_representation = DetectionPrediction(
                 identifier, labels, scores, x_mins / x_scale, y_mins / y_scale, x_maxs / x_scale, y_maxs / y_scale
             )
-            if not self.landmarks_output:
+            if not self.landmarks_output and not self.type_scores_output:
                 results.append(detection_representation)
                 continue
-            landmarks_x_coords = np.array(landmarks_list)[:, :, ::2].reshape(len(landmarks_list), -1) / x_scale
-            landmarks_y_coords = np.array(landmarks_list)[:, :, 1::2].reshape(len(landmarks_list), -1) / y_scale
-            landmarks_representation = FacialLandmarksPrediction(identifier, landmarks_x_coords, landmarks_y_coords)
-            results.append(ContainerPrediction({
-                'detection': detection_representation, 'landmarks_regression': landmarks_representation
-            }))
-
+            representations = {}
+            representations['face_detection'] = detection_representation
+            if self.type_scores_output:
+                representations['mask_detection'] = AttributeDetectionPrediction(
+                    identifier, labels, scores, mask_scores, x_mins / x_scale,
+                    y_mins / y_scale, x_maxs / x_scale, y_maxs / y_scale
+                )
+            if self.landmarks_output:
+                landmarks_x_coords = np.array(landmarks_list)[:, :, ::2].reshape(len(landmarks_list), -1) / x_scale
+                landmarks_y_coords = np.array(landmarks_list)[:, :, 1::2].reshape(len(landmarks_list), -1) / y_scale
+                representations['landmarks_regression'] = FacialLandmarksPrediction(identifier, landmarks_x_coords,
+                                                                                    landmarks_y_coords)
+            results.append(ContainerPrediction(representations))
         return results
 
+    def _get_proposals(self, bbox_deltas, anchor_num, anchors):
+        bbox_deltas = bbox_deltas.transpose((1, 2, 0))
+        bbox_pred_len = bbox_deltas.shape[2] // anchor_num
+        bbox_deltas = bbox_deltas.reshape((-1, bbox_pred_len))
+        proposals = self.bbox_pred(anchors, bbox_deltas)
+        return proposals
+
+    @staticmethod
+    def _get_scores(scores, anchor_num):
+        scores = scores[anchor_num:, :, :]
+        scores = scores.transpose((1, 2, 0)).reshape(-1)
+        return scores
+
+    @staticmethod
+    def _get_mask_scores(type_scores, anchor_num):
+        mask_scores = type_scores[anchor_num * 2:, :, :]
+        mask_scores = mask_scores.transpose((1, 2, 0)).reshape(-1)
+        return mask_scores
+
+    def _get_landmarks(self, landmark_deltas, anchor_num, anchors):
+        landmark_pred_len = landmark_deltas.shape[0] // anchor_num
+        landmark_deltas = landmark_deltas.transpose((1, 2, 0)).reshape((-1, 5, landmark_pred_len // 5))
```

Comment:
It seems, there is also `landmarks_std` for `retinaface-anti-cov` model https://github.com/deepinsight/insightface/blob/3866cd77a6896c934b51ed39e9651b791d78bb57/RetinaFaceAntiCov/retinaface_cov.py#L41. Do you take this into account?

Reply:
landmark_std has been added